src/box/sql/show.c

```diff
@@ -0,0 +1,425 @@
+	sql_desc_append(desc, ",\nCONSTRAINT %s FOREIGN KEY(", constraint_name);
+	sql_xfree(constraint_name);
+	for (uint32_t i = 0; i < cdef->fkey.field_mapping_size; ++i) {
+		const struct tuple_constraint_field_id *field =
+			&cdef->fkey.field_mapping[i].local_field;
+		const char *name = field->name_len != 0 ? field->name :
+				   def->fields[field->id].name;
+		char *field_name = sql_escaped_name_new(name);
+		if (i == 0)
+			sql_desc_append(desc, "%s", field_name);
+		else
+			sql_desc_append(desc, ", %s", field_name);
+		sql_xfree(field_name);
+	}
+
+	char *space_name = sql_escaped_name_new(foreign_space->def->name);
+	sql_desc_append(desc, ") REFERENCES %s(", space_name);
+	sql_xfree(space_name);
+	for (uint32_t i = 0; i < cdef->fkey.field_mapping_size; ++i) {
+		const struct tuple_constraint_field_id *field =
+			&cdef->fkey.field_mapping[i].foreign_field;
+		assert(field->name_len != 0 || field->id < def->field_count);
+		const char *name = field->name_len != 0 ? field->name :
+				   def->fields[field->id].name;
+		char *field_name = sql_escaped_name_new(name);
+		if (i == 0)
+			sql_desc_append(desc, "%s", field_name);
+		else
+			sql_desc_append(desc, ", %s", field_name);
+		sql_xfree(field_name);
+	}
+	sql_desc_append(desc, ")");
+}
+
+/** Add a field check constraint to the statement description. */
+static void
+sql_describe_field_check(struct sql_desc *desc, const char *field_name,
+			 const struct tuple_constraint_def *cdef)
+{
+	assert(cdef->type == CONSTR_FUNC);
+	const struct func *func = func_by_id(cdef->func.id);
+	if (func->def->language != FUNC_LANGUAGE_SQL_EXPR ||
+	    !func_sql_expr_has_single_arg(func, field_name)) {
+		sql_err_append(desc, "Check constraint", cdef->name);
+		return;
+	}
+	char *constraint_name = sql_escaped_name_new(cdef->name);
+	sql_desc_append(desc, " CONSTRAINT %s CHECK(%s)", constraint_name,
+			func->def->body);
+	sql_xfree(constraint_name);
+}
+
+/** Add a tuple check constraint to the statement description. */
+static void
+sql_describe_tuple_check(struct sql_desc *desc,
+			 const struct tuple_constraint_def *cdef)
+{
+	assert(cdef->type == CONSTR_FUNC);
+	const struct func *func = func_by_id(cdef->func.id);
+	if (func->def->language != FUNC_LANGUAGE_SQL_EXPR) {
+		sql_err_append(desc, "Check constraint", cdef->name);
+		return;
+	}
+	char *constraint_name = sql_escaped_name_new(cdef->name);
+	sql_desc_append(desc, ",\nCONSTRAINT %s CHECK(%s)", constraint_name,
+			func->def->body);
+	sql_xfree(constraint_name);
+}
+
+/** Add a field to the statement description. */
+static void
+sql_describe_field(struct sql_desc *desc, const struct field_def *field)
+{
+	char type_str[TYPE_STR_SIZE];
+	const char *type_str_original = field_type_strs[field->type];
+	size_t type_str_len = strlen(type_str_original);
+	assert(type_str_len < TYPE_STR_SIZE);
+	for (size_t i = 0; i < type_str_len; ++i)
+		type_str[i] = toupper(type_str_original[i]);
+	type_str[type_str_len] = '\0';
```

Comment:
We already have `strtolower()` / `strtolowerdup()` in `trivia/util.h`. Maybe worth adding `strtoupper()` / `strtoupperdup()`.